README.md

```diff
@@ -1 +1,168 @@
+device.
+
+The main use case are virtualization workloads inside the pod driven by
+Kubevirt but it can also be used directly with QEMU/libvirt and it might be
+suitable combined with other virtualization backends.
+
+macvtap CNI includes a device plugin to properly expose the macvtap interfaces
+to the pods. A metaplugin such as [Multus](https://github.com/intel/multus-cni)
+gets the name of the interface allocated by the device plugin and is responsible
+to invoke the cni plugin with that name as deviceID.
+
+## Deployment
+
+The device plugin is configured through environment variable `DP_MACVTAP_CONF`.
+The value is a json array and each element of the array is a separate resource
+to be made available:
+
+* `name` (string, required) the name of the resource
+* `master` (string, required) the name of the macvtap lower link
+* `mode` (string, optional, default=bridge) the macvtap operating mode
+* `capacity` (uint, optional, default=100) the capacity of the resource
+
+In the default deployment, this configuration shall be provided through a
+config map, for [example](examples/macvtap-deviceplugin-config.yaml):
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: macvtap-deviceplugin-config
+data:
+  DP_MACVTAP_CONF: |
+    [ {
+        "name" : "dataplane",
+        "master" : "eth0",
+        "mode": "bridge",
+        "capacity" : 50
+    } ]
+```
+
+```bash
+$ kubectl apply -f https://raw.githubusercontent.com/kubevirt/macvtap-cni/master/examples/macvtap-deviceplugin-config.yaml
+configmap "macvtap-deviceplugin-config" created
+```
+
+This configuration will result in up to 50 macvtap interfaces being offered for
+consumption, using eth0 as the lower device, in bridge mode, and under
+resource name `macvtap.network.kubevirt.io/dataplane`.
+
+The macvtap CNI can be deployed using the proposed
+[daemon set](manifests/macvtap.yaml):
+
+```
+$ kubectl apply -f https://raw.githubusercontent.com/kubevirt/macvtap-cni/master/manifests/macvtap.yaml
+daemonset "macvtap-cni" created
+
+$ kubectl get pods
+NAME                                 READY     STATUS    RESTARTS   AGE
+macvtap-cni-745x4                      1/1    Running           0    5m
+```
+
+This will result in the CNI being installed and device plugin running on all
+nodes.
+
+There is also a [template](templates/macvtap.yaml.in) available to parameterize
+the deployment with different configuration options.
+
+## Usage
+
+macvtap CNI is best used by defining a NetworkAttachmentDefinition:
+
+```yaml
+kind: NetworkAttachmentDefinition
+apiVersion: k8s.cni.cncf.io/v1
+metadata:
+  name: dataplane
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: macvtap.network.kubevirt.io/dataplane
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "name": "dataplane",
+      "type": "macvtap-cni"
+      "mtu": 1500
+    }'
+```
+
+The CNI config json allows the following parameters:
+* `name`     (string, required): the name of the network. When used within a
```

Comment:
this sentence is weird, sounds like you didn't finish writing; from where will the name be adopted from when the NAD is omitted ?